tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -244,6 +249,53 @@ def _fetch_resources(self, query):
     def augment(self, resources):
         return resources
 
+    @classmethod
+    def get_urns(cls, resources, project_id=None):
+        """Generate URNs for the resources.
+
+        A Uniform Resource Name (URN) is a URI that identifies a resource by
+        name in a particular namespace. A URN may be used to talk about a
+        resource without implying its location or how to access it.
+
+        The generated URNs can uniquely identify any given resource.
+
+        The generated URN is intended to follow a similar pattern to ARN, but be
+        specific to GCP.
+
+        gcp:<service>:<region>:<project>:<resource-type>/<resource-id>
+
+        If the region is "global" then it is omitted from the URN.
+        """
+        return [cls._get_urn(r, project_id) for r in resources]
+
+    @classmethod
+    def _get_urn(cls, resource, project_id=None) -> str:
+        "Generate an URN for the resource."
+        rt = cls.resource_type
+        region = cls._get_region(resource)
+        if region == "global":
+            region = ""
+        if project_id is None:
```

Comment:
if we have get_region, why not get_project many times this can be inferred from the resource itself